session/pingpong/invoice_tracker.go

```diff
@@ -423,7 +423,7 @@ func (it *InvoiceTracker) Start() error {
 	defer it.revealPromise()
 
 	// give the consumer a second to start up his payments before sending the first request
-	firstSend := time.After(time.Second)
+	firstSend := time.After(3 * time.Second)
```

Comment:
@vkuznecovas 1 second may not be enough for p2p handler to register session-invoice as it happens deep inside session. I was thinking that it could be solved by sending p2p message from consumer to provider notifying that payments are ready, what do you think?

Reply:
Already so many messages, why aren't they enough, "listeners ready" message was especially for that, wasn't it?
So we need just bind to it correctly to remove this sleep

Reply:
Listeners ready message was for provider side. This issue is for consumer side. Provider sends payment-invoice after one second assuming that during this period consumer will set up payments.

Reply:
Ok, I see that this is mostly a problem for e2e tests.

Reply:
Loosing a single invoice is tolerable and should not pose an issue. If it still does - we should fix that.

Reply:
The first session invoice MUST be enforced, this is the only way for the provider to know that the consumer is able to pay, there's no other way to verify. We should be able to drop invoices AFTER the first one, but the first one is critical.

Reply:
#2034

Reply:
If so it should be in session-create response, simple approach to avoid racing

Reply:
Mixing payment logic with session create logic looks like a hack. Would like to decouple payments from session creation since both are not strictly each-other dependent. 

Could we somehow make payment dependent services payment dependent? :) That is if payment is not provided - we don't provide (suspent, put on hold) a service until we get a promise. This way, even if we miss first invoice we still could resume a service after invoice retry. Until then only communication channel works. 

Other approach would be to suspend actual service connect until invoice is received and promise set. But that would be a weak point since consumers might want to abuse such provider behaviour.